packages/api/db/migration/20240504185218_create_soil_amendement_task_products_table.js

```diff
@@ -0,0 +1,573 @@
+    table.enu('elemental_unit', elementalUnits);
+    table.decimal('ammonium');
+    table.decimal('nitrate');
+    table.enu('molecular_compounds_unit', molecularCompoundsUnits);
+    table.check(
+      `(COALESCE(${elements.join(
+        ', ',
+      )}) IS NULL AND elemental_unit IS NULL) OR (COALESCE(${elements.join(
+        ', ',
+      )}) IS NOT NULL AND elemental_unit IS NOT NULL)`,
+      [],
+      'elemental_unit_check',
+    );
+    table.check(
+      `elemental_unit != 'percent' OR (elemental_unit = 'percent' AND (${elements
+        .map((element) => `COALESCE(${element}, 0)`)
+        .join(' + ')}) <= 100)`,
+      [],
+      'elemental_percent_check',
+    );
+    table.check(
+      `(COALESCE(${compounds.join(
+        ', ',
+      )}) IS NULL AND molecular_compounds_unit IS NULL) OR (COALESCE(${compounds.join(
+        ', ',
+      )}) IS NOT NULL AND molecular_compounds_unit IS NOT NULL)`,
+      [],
+      'molecular_compounds_unit_check',
+    );
+  });
+
+  // TODO: Migrate fertilizer table into here?
+  // Insert blank record into soil amendment product
+  await knex.raw(`
+    INSERT INTO soil_amendment_product (product_id)
+    SELECT product_id FROM product
+    WHERE product.type = 'soil_amendment_task'
+  `);
+
+  // Create amendment purpose table
+  await knex.schema.createTable('soil_amendment_purpose', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  // Add purpose types
+  for (const key of amendmentProductPurposeKeys) {
+    await knex('soil_amendment_purpose').insert({
+      key,
+    });
+  }
+
+  // Create amendment task method table
+  await knex.schema.createTable('soil_amendment_method', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  // Add method types
+  for (const key of amendmentMethodKeys) {
+    await knex('soil_amendment_method').insert({
+      key,
+    });
+  }
+
+  // Create new product table
+  await knex.schema.createTable('soil_amendment_task_products', (table) => {
+    table.increments('id').primary();
+    table.integer('task_id').references('task_id').inTable('task').notNullable();
+    // TODO: LF-4246 backfill data if nulls exist on prod and constrain this to notNullable()
+    table.integer('product_id').references('product_id').inTable('product');
+    table.decimal('weight', 36, 12);
+    table.enu('weight_unit', weightUnits);
+    table.decimal('volume', 36, 12);
+    table.enu('volume_unit', volumeUnits);
+    table.enu('application_rate_weight_unit', applicationRateWeightUnits);
+    table.enu('application_rate_volume_unit', applicationRateVolumeUnits);
+    // TODO: LF-4246 backfill data for percent_of_location_amended then make notNullable and defaulted to 100
+    table.decimal('percent_of_location_amended', 36, 12);
+    // TODO: LF-4246 backfill data for total_area_amended_in_ha then make notNullable
+    table.decimal('total_area_amended_in_ha', 36, 12);
```

Comment:
Yup thanks! Just an oversight!

Reply:
Do you want to specify "in m2"? I think we can remove it for consistency :)

Reply:
Tough question! We could probably take it off. It just seemed slightly different since we are not storing units beside like other units. But my point is not a good argument haha

Reply:
added it!